lib/ansible/modules/monitoring/graylog_roles.py

```diff
@@ -0,0 +1,297 @@
+
+    payload = {}
+
+    for key in ['name', 'description', 'permissions', 'read_only']:
+        if module.params[key] is not None:
+            payload[key] = module.params[key]
+
+    response, info = fetch_url(module=module, url=url, headers=json.loads(headers), method='PUT', data=module.jsonify(payload))
+
+    if info['status'] != 200:
+        module.fail_json(msg="Fail: %s" % ("Status: " + str(info['msg']) + ", Message: " + str(info['body'])))
+
+    try:
+        content = to_text(response.read(), errors='surrogate_or_strict')
+    except AttributeError:
+        content = info.pop('body', '')
+
+    return info['status'], info['msg'], content, url
+
+
+def delete(module, base_url, headers):
+
+    url = "/".join([base_url, module.params['name']])
+
+    response, info = fetch_url(module=module, url=url, headers=json.loads(headers), method='DELETE')
+
+    if info['status'] != 204:
+        module.fail_json(msg="Fail: %s" % ("Status: " + str(info['msg']) + ", Message: " + str(info['body'])))
+
+    try:
+        content = to_text(response.read(), errors='surrogate_or_strict')
+    except AttributeError:
+        content = info.pop('body', '')
+
+    return info['status'], info['msg'], content, url
+
+
+def list(module, base_url, headers):
+
+    url = base_url
+
+    response, info = fetch_url(module=module, url=url, headers=json.loads(headers), method='GET')
+
+    if info['status'] != 200:
+        module.fail_json(msg="Fail: %s" % ("Status: " + str(info['msg']) + ", Message: " + str(info['body'])))
+
+    try:
+        content = to_text(response.read(), errors='surrogate_or_strict')
+    except AttributeError:
+        content = info.pop('body', '')
+
+    return info['status'], info['msg'], content, url
+
+
+def get_token(module, endpoint, username, password):
+
+    headers = '{ "Content-Type": "application/json", "X-Requested-By": "Graylog API", "Accept": "application/json" }'
+
+    url = "https://%s/api/system/sessions" % (endpoint)
+
+    payload = {
+        'username': username,
+        'password': password,
+        'host': endpoint
+    }
+
+    response, info = fetch_url(module=module, url=url, headers=json.loads(headers), method='POST', data=module.jsonify(payload))
+
+    if info['status'] != 200:
+        module.fail_json(msg="Fail: %s" % ("Status: " + str(info['msg']) + ", Message: " + str(info['body'])))
+
+    try:
+        content = to_text(response.read(), errors='surrogate_or_strict')
+        session = json.loads(content)
+    except AttributeError:
+        content = info.pop('body', '')
+
+    session_string = session['session_id'] + ":session"
+    session_bytes = session_string.encode('utf-8')
+    session_token = base64.b64encode(session_bytes)
+
+    return session_token
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            endpoint=dict(type='str'),
+            graylog_user=dict(type='str'),
+            graylog_password=dict(type='str', no_log=True),
```

Comment:
looks like there are common options for all graylog modules, Shared code are located in lib/ansible/module_utils (note that this must not be GPL here). But this can be done later on. Not a blocker, just a hint.